services/consensusalgo/leanhelix/algorithm.go

```diff
@@ -0,0 +1,91 @@
+package leanhelix
+
+import (
+	"github.com/orbs-network/orbs-network-go/services/blockstorage"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+func (s *service) leaderProposeNextBlockIfNeeded() error {
```

Comment:
this is very nice, I like the properly named functions. Please add `reporting.info` calls in all functions so that we have better instrumentation of the algorithm in action